rofication/_server.py

```diff
@@ -44,7 +44,14 @@ def delete_all(self, application: str) -> None:
 
     def list(self, fp: TextIO) -> None:
         with self.server.queue.lock:
-            json.dump(list(self.server.queue), fp, default=Notification.asdict)
+            messages = list(self.server.queue)
+            sort_fields = resources.notify_sort_by.fetch().split()
+            sort_fields.reverse()
+            for sort_field in sort_fields:
+                reverse = sort_field.startswith("!")
+                sort_field = sort_field.lstrip("!")
+                messages.sort(key=lambda message: message.asdict()[sort_field], reverse=reverse )
```

Comment:
This will crash if there's a typo in the field and it doesn't exist in the message.

A possibility would be, rather than crashing, returning a list with a single message reporting the error (something like "Field XX does not exist, check i3xrocks_notify_sort_by in XResource. Valid fields are....)

Reply:
I updated the code. Now if an incorrect field name is used, a message indicating that  is added at the top of the list and the rest of the sorting is done normally.